### PR TITLE
Document enhanced tracing in Studio via OTel

### PR DESCRIPTION
**Pull request by timbotnik**
Officially document enhanced tracing via OTel.  This is currently marked experimental and we've started to get some usage of this feature.  Time to make it more visible.

Fixes [#712](https://apollographql.atlassian.net/browse/ROUTER-712)

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [x] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

**Review by bnjjj — APPROVED** [review]

File: docs/source/configuration/telemetry/apollo-telemetry.mdx [hunk 1]
```diff
@@ -72,6 +72,70 @@ telemetry:
...
+        sampler: 0.01
+```
+
+OTel traces sent to Studio will not necessarily be identical to the ones sent to 3rd Party APM tools via OTLP:
+
+- Only specific OTLP attributes will be included for parity with what is provided in legacy traces today. This ensures that data privacy
+  is maintained in an equivalent manner. The existing Router configuration options for Apollo telemetry will continue to function
+  with OTLP traces, such as forwarding of GraphQL errors, headers, and variables.
+- Some features of OTLP traces may only be available in Studio and not in 3rd Party APM tools (e.g. resolver-level timing information from
+  [Federated Tracing](../../federation/metrics/#enabling-federated-tracing)).
```
Comment (shorgi) [inline docs/source/configuration/telemetry/apollo-telemetry.mdx:127]:
```suggestion
OTLP traces sent to Studio aren't necessarily identical to ones sent to third-party APM tools via OTLP:

- Only specific OTLP attributes are included for parity with legacy traces today. This ensures that data privacy is maintained in an equivalent manner. Existing router configuration options for Apollo telemetry will continue to function with OTLP traces, including forwarding of GraphQL errors, headers, and variables.
- Some features of OTLP traces are available only in Studio and not in third-party APM tools, such as resolver-level timing information from [Federated Tracing](../../federation/metrics/#enabling-federated-tracing)
```

File: docs/source/configuration/telemetry/apollo-telemetry.mdx [hunk 2]
```diff
@@ -72,6 +72,70 @@ telemetry:
...
+
+<Note>
+
+This change results in using a new wire protocol for traces, and some users may experience an increase in tracing traffic
+to GraphOS Studio due to the additional detail being captured. In exceptional situations it may be necessary to send fewer traces.
+This can be achieved via sending fewer traces (`telemetry.exporters.tracing.common.sampler`) or as a last resort, falling back
+to the old protocol via `telemetry.apollo.otlp_tracing_sampler` to send fewer OTLP traces or fully disable them.  
+Any performance regressions due to the new tracing protocol should also be reported to the Apollo support team.
```
Comment (shorgi) [inline docs/source/configuration/telemetry/apollo-telemetry.mdx:135]:
```suggestion
You may experience an increase in tracing traffic sent to GraphOS Studio due to the additional detail captured by the new wire protocol. In exceptional situations, you may need to send fewer traces.

To send fewer traces, configure `telemetry.exporters.tracing.common.sampler` or revert to the old protocol via `telemetry.apollo.otlp_tracing_sampler` to send fewer OTLP traces or to disable them.
  
For performance regressions due to the new tracing protocol, you should report them to the [Apollo support team](https://www.apollographql.com/support).
```

File: docs/source/configuration/telemetry/apollo-telemetry.mdx [hunk 3]
```diff
@@ -72,6 +72,71 @@ telemetry:
...
+- `0.0 - 1.0` (used for testing): the ratio of traces to send via OTLP (0.4 = 40% OTLP / 60% legacy)
+
+This sampler is applied after the common tracing sampler.
+
+#### Example configuration 
+
+An example configuration that samples 1% of traces and sends all traces via OTLP:
+
+
+```yaml
```
Comment (bnjjj) [inline docs/source/configuration/telemetry/apollo-telemetry.mdx:112]:
```suggestion
```yaml title="router.config.yaml"
```

To make sure we're validating this configuration in our tests.

**Review by bnjjj — APPROVED** [review]

**Comment by Meschreiber** [comment]
!docs preview

**Pull request merged**